source/develop/release-management/features/virt/high-performance-vm.html.md

```diff
@@ -93,60 +91,67 @@ Displayed in 'Resource Allocation' side-tab of the VM dialog.
 
-In case the CPU Pinning topology is not set upon VM saving,  a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the VM to a host (by selecting the "Start running on: Specific Host" in "Host" side-tab) and verify if VM configuration fits the host configuration (number of cores per socket,  threads per core and number of sockets should match).
+In case the CPU Pinning topology is not set upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the VM to a host (by selecting the "Start running on: Specific Host" in "Host" side-tab) and verify if VM configuration fits the host configuration (number of cores per socket,  threads per core and number of sockets should match).
 
-Ideally the CPU pinning should be done automatically and not manually by the user, But in first phase we will not support it.  The automatic pinning can be done by pinning to the CPU that is responsible for the IO in the host (and don't present this CPU to the guest) or alternatively pinned to CPU#0 as this is typically the one.  
-
-In addition we will suggest to set the emulatorpin in libvirt xml.
+Ideally the CPU pinning should be done automatically and not manually by the user, But in first phase we will not support it.  
 
 #### **Set the I/O Thread pinning topology**
 This is currently not implemented. 
-Should be added to the 'Resource Allocation' side-tab of the VM dialog. It will be implemented by setting  the iothreadpin in libvirt xml accordingly
+Should be added to the 'Resource Allocation' side-tab of the VM dialog. It will be implemented by setting the iothreadpin in libvirt xml accordingly (requires VDSM changes).
 
-This feature should be implemented the same as the CPU pinning is implemented.
+This feature should be implemented the same as the CPU pinning is implemented. 
+Ideally the the I/O Thread pinning should be done automatically and not manually by the user, But in first phase we will not support it.  The automatic pinning can be done by pinning to the CPU that is responsible for the IO in the host (and don't present this CPU to the guest) or alternatively pinned to CPU#0 as this is typically the one. 
 
-In case the I/O Thread pinning topology is not set upon VM saving,  a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the I/O thread to a host and verify it fits the host configuration.
+In case the I/O Thread pinning topology is not set upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the I/O thread to a host and verify it fits the host configuration.
 Consider verifying if the  I/O is bound to specific CPU on host level and chose that CPU for pinning the I/O Thread.
 
+In addition, we will suggest to set the emulatorpin to be pinned to the same CPU(s) as the I/O Thread. Should consider how the user will do that (using a hook or via VDSM).
```

Comment:
It might make more sense to actually pin it the the next free CPU (leaving first 2 CPUs per numa node for IO/emu).